pennylane/pauli/dla/lie_closure.py

```diff
@@ -422,12 +422,11 @@ def _check_independence(M, pauli_sentence, pw_to_idx, rank, num_pw, tol):
             M[pw_to_idx[pw], rank] = value
 
         # Check if new vector is linearly dependent on the current basis
-        v = M[:, -1].copy()  # remove copy to normalize M
+        v = M[:, -1]  # .copy()  # remove copy to normalize M
         v /= np.linalg.norm(v)
         A = M[:, :-1]
-        v = v - A @ qml.math.linalg.solve(qml.math.conj(A.T) @ A, A.conj().T) @ v
 
-        if np.linalg.norm(v) > tol:
+        if np.linalg.norm(A @ qml.math.linalg.pinv(A) @ v - v) > tol:
```

Comment:
If A has linearly independent columns, `pinv` should be equivalent to the above, but `pinv` is more general and robust, using an SVD to handle close-to-zero singular values. I thought A's columns were supposed to remain linearly independent throughout. Are we sure this is not happening? Or does it not matter? Or are we reaching the limits or numerical precision as we try to maintain a basis of the original vectors (and hence we're not allowed to orthogonalize)?

Reply:
I think this comment is spot-on. 

The cases where this was failing were linear combinations of Paulis that lead to huge normalizations of high-order commutators. This causes numerical instability, I believe, although just adding (some approximate) normalization did not fix the original issue. 

I'm wondering whether the fix in this PR still catches all linearly independent elements of the DLA correctly, will add an exemplary test!

I think there is something to be understood, still, regarding the causing issue :thinking: 

@Qottmann do you have any adidtional insight?

Reply:
Yeah honestly still a bit puzzled why this breaks for larger n and in particular when invoking sums of paulis. As David mentions, there are often terms with really large numerical values, though just normalizing stuff doesnt do the trick. Another point is that the terms can become quite large, but still not sure how that should cause numerical instabilities.

Reply:
I think I actually havent fully understood what we are doing in this test and just took it for ghospel 😅 
Now that I think about it, can we not just check `np.min(singular_values_of_A) > tol`? I.e. as soon as one of the singular values becomes $0$ we have linear dependence?

Reply:
I would not expect normalization to help here, but performing orthogonalization at the cost of losing track of the original vectors should help. If that doesn't cut it though, then the problem may simply be very badly conditioned. Without iterative orthogonalization, I'm afraid we will quickly lose the ability to distinguish between linearly dependent and independent vectors not matter what the test is. Could we explore this avenue and consider adding an option if it works?

Reply:
>I think I actually havent fully understood what we are doing in this test and just took it for ghospel 😅
Now that I think about it, can we not just check np.min(singular_values_of_A) > tol? I.e. as soon as one of the singular values becomes 0 we have linear dependence? (and isnt that equivalent to computing the rank?)

Hmhm, good question, that could potentially work, no?
Didn't we have a version that just computed the rank directly, at some point? Why did we move away from that?

Reply:
>Could we explore this avenue and consider adding an option if it works?

I'm definitely down for an option that enables numerical stability at the cost of loosing track of original input vectors, if users can control it.
I must be the mean guy here and say I don't have time for this on the side, though :grimacing: I personally would need to plan in some dedicated time for exploring this avenue.

Reply:
> Didn't we have a version that just computed the rank directly, at some point? Why did we move away from that?

That is done in `__eq__` for comparing two vspaces

> Could we explore this avenue and consider adding an option if it works?

Same as David, though if this becomes a blocker for my research I might try something in that direction.
It could be triggered by a flag `orthogonalize=True/False` or so. If I went down that road, would the check be the same, just more stable? Or would if also simplify the test?